File: Solution/WebSolution/Java/AssessmentAPI/src/main/java/com/transflower/tflassessment/controllers/EvaluationCriteriaController.java
package com.transflower.tflassessment.controllers;

import java.util.concurrent.CompletableFuture;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.transflower.tflassessment.entities.EvaluationCriteria;
import com.transflower.tflassessment.services.EvaluationCriteriaService;


@RestController
@RequestMapping("/api/criteria")
public class EvaluationCriteriaController {

    @Autowired
    private EvaluationCriteriaService svc;

    @PostMapping("/insert")
    public CompletableFuture<Boolean> insertCriteria(@RequestBody EvaluationCriteria criteria) {
        return svc.insertCriteria(criteria);
    }

   @PutMapping("{id}/subjects/{subjectId}")
    public CompletableFuture<Boolean> updateSubject(@PathVariable("id") int id, @PathVariable ("subjectId")int subjectId) {
    return svc.updateSubject(id, subjectId);
}

    @PutMapping("{id}/questions/{evaluationCriteriaId}")
    public CompletableFuture<Boolean> updateCriteria(@PathVariable("id")int id, @PathVariable("evaluationCriteriaId") int evaluationCriteriaId) {
        return svc.updateCriteria(id, evaluationCriteriaId);
    }
   
}

File: Solution/WebSolution/Dotnet/AssessmentPortal/Monolithic/Monolithic_Controller/AssessmentAPI/Controllers/EvaluationCriteriaController.cs
using System.Diagnostics;
using Microsoft.AspNetCore.Mvc;
using Transflower.TFLAssessment.Entities;
using Transflower.TFLAssessment.Services.Interfaces;

[ApiController]
[Route("api/criteria")]
public class EvaluationCriteriaController : ControllerBase
{   

    private readonly ILogger <EvaluationCriteriaController> _logger;
    private readonly IEvaluationCriteriaService _svc;
    
    public EvaluationCriteriaController(IEvaluationCriteriaService service, ILogger<EvaluationCriteriaController> logger)
    {
        _svc = service;
        _logger= logger;
    } 
   
   
    [HttpPut("{evaluationCriteriaId}/questions/{questionId}")]
    public async Task<IActionResult> UpdateCriteria(int evaluationCriteriaId, int questionId)
    {
        bool status = await _svc.UpdateCriteria(evaluationCriteriaId, questionId);
        if (!status){
            _logger.LogError("Failed to update criteria for evaluationCriteriaId: {EvaluationCriteriaId} and questionId: {QuestionId}", evaluationCriteriaId, questionId);
            return BadRequest("Failed to update criteria.");    
        }
         _logger.LogInformation("Log Generated to Insert candidate answers of the test");
        return Ok(status);
    }
    
    [HttpPut("{id}/subjects/{subjectId}")]
    public async Task<IActionResult> UpdateSubject(int id, int subjectId)
    {
        bool status = await _svc.UpdateSubject(id, subjectId);
        _logger.LogInformation("Log Generated For update subject");
        return Ok(status);
    }

    [HttpPost]
    public async Task<IActionResult> InsertCriteria(EvaluationCriteria criteria)
    {  
        bool status = await _svc.InsertCriteria(criteria);
         _logger.LogInformation("Log Generated For Insert Criteria ");
        return Ok(status);
    }

    [HttpGet("{id}")]
    public async Task<IActionResult> GetCriteriaBySubjectId(int id)
    {
        List<EvaluationCriteria> evaluationCriteria = await _svc.GetCriteriaBySubjectId(id);
        return Ok(evaluationCriteria);
    }

    [HttpGet("questioncount/{id}")]
    public async Task<IActionResult> getCriteriaQuestionCount(int id)
    {
        List<CriteriaQuestionCount> evaluationCriteriaCount = await _svc.getCriteriaQuestionCount(id);
        return Ok(evaluationCriteriaCount);
    }
}
